relion/constants.py

```diff
@@ -41,7 +41,7 @@
 MASK_FILL_ZERO = 0
 MASK_FILL_NOISE = 1
 
-ANGULAR_SAMPLING_LIST = ['30', '15', '7.5', '3.7', '1.8', '0.9', '0.5', '0.2', '0.1']
+ANGULAR_SAMPLING_LIST = ['30', '15', '7.5', '3.7', '1.8', '0.9', '0.5', '0.2', '0.1', '0.06', '0.03', '0.01', '0.007', '0.004']
```

Comment:
This is not accepted. "the reason is again localrec, the very small values are an indirect way ti tell relion that it should not refine the angles"  is not a valid reason, if you dont want to refine angles, switch off the alignment.

Reply:
Let me try to explain why we need this extra angular steps.  In my last processing  assuming symmetry i1, relion_auto_refine used in the last iterations an angular_sampling_step= 0.03.  OK now I think that one of the vertex of my virus is different and I want to refine using c1. Of course I may rerun relion_refine from the beginning but a better and faster alternative is to use the  "--relax_sym i1" and  "--sigma_ang 0.00001" flags.  For each set of angles "--relax_sym i1" will try the 59 angles related by symmetry and  "--sigma_ang 0.00001" (when used together with "consider alignment as priors") will stop any modification in the angles that is not one of the 59 symmetry related positions.  

What value should I use as initial sampling rate? If I use something larger than 0.03 I will lose information. Let us imagine that I use 0.1 as the angular_sampling_rate, then I will loose information because relion cannot assign arbitrary angles to the projections, it can only choose between the set of angles  produced by the algorithm with the coarser angular_sampling_rate=0.1. Let us imagine that I choose angular_sampling_rate=0.03, this sampling is not good enough, for the first angle of the 60 angles related by symmetry it will be OK but as soon as I try any of the other 59 angles the value will not exist. The solution would be to use an angular_sampling_rate half the size that the one use for relion in the last iteration, that is, 0.01. Just to be in the save side we may include  the next step that is 0.007. If you wish I will be happy to drop 0.004.

Reply:
@rmarabini I see, you are trying to push changes that have nothing to do with this subtraction PR. sigma_ang (within scipion) is only used in local 3d classification an not local 3d refinement (for which you use healpix order for initial and local sampling), but surely you can type anything in the advanced params (btw, the last parameter value will be used if you specified the same param more than once). 

Hi, @JuhaHuiskonen, could you confirm if this is the correct way to do it? I have not used relax_sym in Relion.

Reply:
@rmarabini is correct.  When the --relax_sym option is used, it is a type of alignment.  So --skip_align cannot be used at the same time.  In most cases, one wants to make this symmetry relaxed alignment very local, i.e. test only one orientation for each symmetry related direction (just the 60 directions for I symmetry and nothing around these). This can be achieved with --sigma_ang 0.001 or some other very small value. In addition, a very high global HP value is needed to not lose accuracy. In other words, the library or directions where the symmetry mates are being selected from needs to be sampled fine enough to not approximate the input angles. (These will not be refined further in this example, as you are just selecting one of the symmetry mates.) For this reason the global HP value should be at least as high as the final HP value of the earlier alignment. Also note that in 3D auto-refine, even higher local HP value is needed so that RELION never reaches the point where it would change to its own local alignment logic. (This would change the local angular search space larger than set by --sigma_ang and suddenly one starts to get more  directions than intended.)

Reply:
Thank you both! I've learned something new.